hail/python/hailtop/aiogoogle/client/storage_client.py

```diff
@@ -242,13 +242,31 @@ async def _listfiles_flat(self, bucket: str, name: str) -> AsyncIterator[FileLis
                 for item in page['items']:
                     yield GoogleStorageFileListEntry(f'gs://{bucket}/{item["name"]}', item)
 
-    def listfiles(self, url: str, recursive: bool = False) -> AsyncIterator[FileListEntry]:
+    async def listfiles(self, url: str, recursive: bool = False) -> AsyncIterator[FileListEntry]:
         bucket, name = self._get_bucket_name(url)
         if not name.endswith('/'):
             name = f'{name}/'
+
         if recursive:
-            return self._listfiles_recursive(bucket, name)
-        return self._listfiles_flat(bucket, name)
+            it = self._listfiles_recursive(bucket, name)
+        else:
+            it = self._listfiles_flat(bucket, name)
+
+        it = it.__aiter__()
+        try:
+            n = await it.__anext__()
```

Comment:
I thought `n` was a number. Can you change the variable name?

Reply:
Done.